### PR TITLE
Refactor lib/access.js tests

### LINKED ISSUE #1474: Add lib/access.js unit tests
- Convert `lib/access.js` tests cases to unit tests
- Any test cases that relies on integration can still live in `test/tap/`

### PR DESCRIPTION
**Pull request by ruyadorno**
- Refactored lib/access.js to async/await
- Moved tap tests to unit tests under tests/lib/

fix #1474

### CODE REVIEW

**Comment by ruyadorno** [comment]
oh, also replaced `read-package-json` with `read-package-json-fast`

**Comment by isaacs** [comment]
How hard would it be to get the coverage of `lib/access.js` up to 100% here?  It'd be nice to maintain full coverage on the new set of unit tests.

File: lib/access.js [hunk 1]
```diff
@@ -27,23 +25,50 @@ access.usage = usage(
...
+  '2fa-required',
+  '2fa-not-required'
 ]
 
-function UsageError (msg = '') {
-  throw Object.assign(new Error(
+const UsageError = (msg = '') =>
+  Object.assign(new Error(
     (msg ? `\nUsage: ${msg}\n\n` : '') +
     access.usage
```
Comment (isaacs) [inline lib/access.js:43]: this should just be `usage`, since it's no longer assigned to the `access` fn.

**Comment by isaacs** [comment]
c971179d1b6f66c18e322dccb7a2c0c07218ddf9

**Pull request closed without merging**